### PR TITLE
[UPD] migrated base_phone to Odoo 9.0

### PR DESCRIPTION
**Pull request by jmalinens**

### CODE REVIEW

File: asterisk_click2dial/asterisk_click2dial.py [hunk 1]
```diff
@@ -18,9 +18,8 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-
-from openerp.osv import fields, orm
-from openerp.tools.translate import _
+from openerp.exceptions import ValidationError
+from openerp import models, api, fields, api, _
```
Comment (Garamotte) [inline asterisk_click2dial/asterisk_click2dial.py:22]: You should imoprt api only once.

File: asterisk_click2dial/asterisk_click2dial.py [hunk 2]
```diff
@@ -107,88 +110,64 @@ def _check_validity(self, cr, uid, ids):
...
             if not asterisk_server_ids:
-                raise orm.except_orm(
-                    _('Error:'),
-                    _("No Asterisk server configured for the company '%s'.")
+                raise ValidationError(_("No Asterisk server configured for the company '%s'.")
                     % user.company_id.name)
             else:
-                ast_server = self.browse(
-                    cr, uid, asterisk_server_ids[0], context=context)
+                ast_server = self.browse(asterisk_server_ids[0])
```
Comment (Garamotte) [inline asterisk_click2dial/asterisk_click2dial.py:149]: The search method returns a recordset in new API, not a list of ids.

File: asterisk_click2dial/asterisk_click2dial.py [hunk 3]
```diff
@@ -378,36 +342,22 @@ def _check_validity(self, cr, uid, ids):
...
-
-
-class PhoneCommon(orm.AbstractModel):
+class PhoneCommon(models.AbstractModel):
     _inherit = 'phone.common'
 
-    def click2dial(self, cr, uid, erp_number, context=None):
-        res = super(PhoneCommon, self).click2dial(
-            cr, uid, erp_number, context=context)
+    def click2dial(self, erp_number):
```
Comment (Garamotte) [inline asterisk_click2dial/asterisk_click2dial.py:353]: Shouldn't you add @api.model decorator ?

File: base_phone/base_phone.py [hunk 4]
```diff
@@ -41,33 +40,29 @@ def _generic_reformat_phonenumbers(
             'Wrong self._partner_field'
         assert isinstance(self._phone_fields, list),\
             'self._phone_fields must be a list'
-        if context is None:
-            context = {}
         if ids and isinstance(ids, (int, long)):
             ids = [ids]
         if any([vals.get(field) for field in self._phone_fields]):
-            user = self.pool['res.users'].browse(cr, uid, uid, context=context)
+            user = self.env['res.users'].browse(self.env.uid)
```
Comment (Garamotte) [inline base_phone/base_phone.py:46]: user = self.env.user

File: base_phone/wizard/reformat_all_phonenumbers.py [hunk 5]
```diff
@@ -58,18 +58,16 @@ def run_reformat_all_phonenumbers(self, cr, uid, ids, context=None):
                 domain = ['|', ('active', '=', True), ('active', '=', False)]
             else:
                 domain = []
-            all_ids = obj.search(cr, uid, domain, context=context)
-            for entry in obj.read(
-                    cr, uid, all_ids, fields, context=context):
+            all_ids = obj.search(domain)
+            for entry in obj.read(all_ids, fields):
```
Comment (Garamotte) [inline base_phone/wizard/reformat_all_phonenumbers.py:62]: Search method returns a recordset in new API.
You can use the recordsets directly in the for loop, without using the read method.

**Comment by pedrobaeza** [comment]
Superseeded by https://github.com/OCA/connector-telephony/pull/83, so I close.

**Pull request closed without merging**